src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -304,10 +308,18 @@ class LedgerImpl(
       Nil
   }
 
-  def executeBlock(block: Block): Either[BlockExecutionError, Seq[Receipt]] = {
+  /**
+    * Executes a block
+    *
+    * @param alreadyValidated should we skip pre-execution validation (if the block has already been validated,
+    *                         eg. in the importBlock method)
+    */
+  def executeBlock(block: Block, alreadyValidated: Boolean = false): Either[BlockExecutionError, Seq[Receipt]] = {
```

Comment:
this seems to be getting a bit tangled up (and I don't mean just this method but Ledger in general)

Reply:
[EC-313](https://iohk.myjetbrains.com/youtrack/issue/EC-313)